File: packages/client/src/namespaces/terminal.ts
//! `terminal.*` client namespace. Wave 2 owner: A (M1.7 display-only terminal).
//!
//! TODO(A): type `list`, `attach`, `write` and `resize` against tethys-api;
//! every wrapper is still `call<void>()`.

import type { Call } from "../transport";

export function terminalNamespace(call: Call) {
  return {
    list: () => call<void>("terminal_list"),
    attach: () => call<void>("terminal_attach"),
    write: () => call<void>("terminal_write"),
    resize: () => call<void>("terminal_resize"),
  };
}

File: crates/tethys-api/tests/stubs/terminal.rs
use tethys_api::TerminalApi;

use super::{assert_unimplemented, MinimalApi};

#[tokio::test]
async fn terminal_defaults_return_unimplemented() {
    let api = MinimalApi;
    assert_unimplemented("terminal.list", api.terminal_list().await);
    assert_unimplemented("terminal.attach", api.terminal_attach().await);
    assert_unimplemented("terminal.write", api.terminal_write().await);
    assert_unimplemented("terminal.resize", api.terminal_resize().await);
}
